workginCopyFileService: only one event for multiple operations

isidorn opened the pull request:
@bpasero for some weird reason I could not assign you to reviewer here.

This PR tackles the feedback from this comment https://github.com/microsoft/vscode/issues/111867#issuecomment-755300132

I did not merge this in to master because this breaks the following integration test https://github.com/microsoft/vscode/blob/c6cc7d614012c1073248be37db2a012746390008/extensions/vscode-api-tests/src/singlefolder-tests/workspace.event.test.ts#L23

@jrieken not sure if this needs to be adopted in some other place

bpasero commented on src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts:
```diff
@@ -272,37 +272,40 @@ export class WorkingCopyFileService extends Disposable implements IWorkingCopyFi
-			if (validateCreate instanceof Error) {
-				throw validateCreate;
-			}
+			await Promise.all(operations.map(async operation => {
```
I think you could also just await all those `canCreateFile` at once and check for the first `Error` in that array and throw that? Since in the code below you also bulk-create, I think it would be consistent.

bpasero reviewed (approved):
LGTM with one suggestion, thanks!

jrieken commented:
> I did not merge this in to master because this breaks the following integration test

jrieken commented:
> I did not merge this in to master because this breaks the following integration test

That's an older test and it shouldn't need adoption. When it fails it means that either onWill or onDid event didn't fire properly.

isidorn commented:
@bpasero I have addressed your feedback
@jrieken I have tackled the test. The issue was that with my changes I was firing change events even if no `operations` have been passed in. I have changed that, and the tests pass now.

Thus merging in.
